direr.go

```diff
@@ -16,19 +19,81 @@ func TestDirer(t *testing.T, store types.Storager) {
 			So(ok, ShouldBeTrue)
 		})
 
-		Convey("When CreateDir", func() {
+		Convey("When CreateDir with path not end with `/`", func() {
 			d, _ := store.(types.Direr)
 
 			path := uuid.New().String()
 			o, err := d.CreateDir(path)
 
 			defer func() {
-				err := store.Delete(path)
+				err := store.Delete(path, pairs.WithObjectMode(types.ModeDir))
 				if err != nil {
 					t.Error(err)
 				}
 			}()
 
+			Convey("The first returning error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			o, err = d.CreateDir(path)
+			Convey("The second returning error also should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("The Object Mode should be dir", func() {
+				// Dir object's mode must be Dir.
+				So(o.Mode.IsDir(), ShouldBeTrue)
+			})
+		})
+
+		Convey("When CreateDir with path end with `/`", func() {
```

Comment:
For now, we will not check the error when the user inputs `test/`, just ignore it.

Reply:
Does it mean the `path` is valid for services and the path check will be added after https://github.com/beyondstorage/specs/issues/95 implemented? Also we don't need to check the `path` in services?

Reply:
Yes.